src/Component/Page/Home.purs

```diff
@@ -0,0 +1,208 @@
+module Component.Page.Home where
+
+import Prelude
+
+import Api.Endpoint (ArticleParams, noArticleParams)
+import AppM (Env)
+import Capability.LogMessages (class LogMessages)
+import Capability.ManageResource (class ManageResource, getArticles, getTags)
+import Capability.Navigate (class Navigate, navigate)
+import Capability.Now (class Now)
+import Component.Classes as CC
+import Component.HTML.ArticleList (articleList)
+import Component.HTML.Footer (footer)
+import Component.HTML.Header (header)
+import Control.Monad.Reader (class MonadAsk)
+import Data.Article (Article)
+import Data.Const (Const)
+import Data.Either (either)
+import Data.Maybe (Maybe(..))
+import Data.Route (Route(..))
+import Effect.Aff.Class (class MonadAff)
+import Halogen as H
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties as HP
+import Halogen.Themes.Bootstrap4 as HB
+import Network.RemoteData (RemoteData(..))
+import Web.Event.Event (Event, preventDefault)
+import Web.UIEvent.MouseEvent.EventTypes (click)
+
+type State =
+  { tags :: RemoteData String (Array String)
+  , articles :: RemoteData String (Array Article)
+  , tab :: Tab
+  }
+
+data Tab
+  = FeedTab
+  | GlobalTab
+  | TagTab String
+
+data Query a
+  = Init a
+  | LoadTags a
+  | Navigate Route a
+  | ShowTab Tab Event a
+  | LoadArticles ArticleParams a
+
+component
+  :: forall m
+   . MonadAff m
+  => MonadAsk Env m
+  => Now m
+  => LogMessages m
+  => ManageResource m
+  => Navigate m
+  => H.Component HH.HTML Query Unit Void m
+component =
+  H.lifecycleParentComponent
+    { initialState: const { tags: NotAsked, articles: NotAsked, tab: GlobalTab } 
+    , render
+    , eval
+    , receiver: const Nothing
+    , initializer: Just $ Init unit
+    , finalizer: Nothing
+    }
+
+  where 
+
+  eval :: Query ~> H.ParentDSL State Query (Const Void) Void Void m
+  eval = case _ of
+    Init a -> do
+      handle <- H.fork $ eval $ LoadTags a
+      eval $ LoadArticles noArticleParams a
+    LoadTags a -> do
+      H.modify_ _ { tags = Loading}
+      tags <- H.lift getTags
+      H.modify_ _ { tags = either Failure Success tags }
+      pure a
+    LoadArticles params a -> do
+      H.modify_ _ { articles = Loading }
+      articles <- H.lift $ getArticles params
+      H.modify_ _ { articles = either Failure Success articles }
+      pure a      
+    Navigate route a -> do
```

Comment:
I haven't run this code yet, but I _think_ these are going to need a `preventDefault`, too, because the provided HTML has all these navigation links in things like the header and footer set as `a` tags. That can be solved by having this take the event and a route.

Reply:
It seems a least on firefox that if the `a` tag does not have an `href` there is no default, in our case anyway it may be best to use `button` since we don't have any meaningful url to use.

Reply:
Yes, let's switch to `<button>`.